pkg/config/registry/registry.go

```diff
@@ -0,0 +1,51 @@
+// SPDX-FileCopyrightText: 2020-present Open Networking Foundation <info@opennetworking.org>
+//
+// SPDX-License-Identifier: LicenseRef-ONF-Member-1.0
+
+package registry
+
+import (
+	"encoding/json"
+
+	"github.com/onosproject/onos-lib-go/pkg/logging"
+	"github.com/onosproject/onos-ric-sdk-go/pkg/config/agent"
+	"github.com/onosproject/onos-ric-sdk-go/pkg/config/configurable"
+)
+
+var log = logging.GetLogger("registry")
+
+type RegisterRequest struct {
+	server *agent.Server
+}
+
+type RegisterResponse struct {
+	AgentService agent.GnmiService
+	Config       *configurable.ConfigStore
+}
+
+func RegisterConfigurable(c *RegisterRequest) (RegisterResponse, error) {
```

Comment:
This function will be called by the xApp

Reply:
IMO, the app should not have to know anything about the store or the agent details, no? The app should just call `Register(Configurable)` and that's it. If there are other provisioning details, those should come on the side somewhere. Remember, the idea is that the SDK *absorbs* as much complexity as possible.

Reply:
There are two pieces that I expose: One the service to be added to the NB of the App and the other one is just the configuration which is technically a wrapper around map[string]interface. Our codes in the subsystems and the apps are structured in a way that we  add NB services in the manager of a system and start the server there. We can start the server in the RegisterConfigurable and does not expose anything but what would be the service address then? If we add it to the NB of the xApp itself, then the name of app service with port 5150 will be used for gnmi agent and can be added as an entity to the topo. If an xApp is not responsible for adding NB services to its NB and starting the server then there should be a mechanism for all of them not just gnmi agent and I don't know what would be that mechansim and open to suggestions. 

It is not clear for me if we don't expose a data structure in memory how does xApp can access to its config? When an xApp register itself, it should have access to its config even if it is just a default config. Any ideas on this?

Reply:
@tomikazi 
Perhaps I have an idea for the hiding the agent service thing even now. Will get back to you on that.

Reply:
@tomikazi 
So I think I addressed your comment about starting the server in RegisterConfigurbale. IANA reserverd port 9339 for gNMI so I think it would be better start the server on that port and update the helm chart to expose that port. By doing that we can keep port 5150 for other purposes in the NB of the xApp. What do you think?

Reply:
And I am thinking If I can hide more details about another comment .....

Reply:
@tomikazi 
I made some more changes and decoupled config store with what actually app will have access. If you have more comments, please let me know.